### PR TITLE
Ensure that DiagnosticSuppressors are always executed in the IDE for …

### LINKED ISSUE #41573: DiagnosticSuppressor is ignored in FullSolutionAnalysis
**Version Used**: 3.3.1, 3.4.0

**Steps to Reproduce**:

1. Write a [DiagnosticSuppressor](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.diagnostics.diagnosticsuppressor)(the Suppressor I wrote is for IDE0051)
2. Enable Full Solution Analysis

**Expected Behavior**:
Suppressor always supresses the Diagnostic.

**Actual Behavior**:
Suppressor only supresses the Diagnostic if the document is open in Visual Studio. When the file is closed the Diagnostic is not suppressed and the Supressormethod `ReportSuppressions` is not called.

### PR DESCRIPTION
**Pull request by mavasani**
…all backgroung analysis scopes

Fixes #41573

### CODE REVIEW

File: src/EditorFeatures/Test/Microsoft.CodeAnalysis.EditorFeatures.UnitTests.csproj [hunk 1]
```diff
@@ -65,6 +65,7 @@
     <PackageReference Include="Microsoft.VisualStudio.Threading" Version="$(MicrosoftVisualStudioThreadingVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.Utilities" Version="$(MicrosoftVisualStudioUtilitiesVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.Validation" Version="$(MicrosoftVisualStudioValidationVersion)" />
+    <PackageReference Include="Xunit.Combinatorial" Version="$(XunitCombinatorialVersion)" PrivateAssets="all" />
```
Comment (sharwell) [inline src/EditorFeatures/Test/Microsoft.CodeAnalysis.EditorFeatures.UnitTests.csproj]:
```suggestion
    <PackageReference Include="Xunit.Combinatorial" Version="$(XunitCombinatorialVersion)" />
```

**Comment by mavasani** [comment]
Blocked until #41713 is fixed.

**Comment by mavasani** [comment]
@sharwell This PR is no longer blocked, fix for #41713 has been merged.

File: src/EditorFeatures/Test/Microsoft.CodeAnalysis.EditorFeatures.UnitTests.csproj [hunk 2]
```diff
@@ -65,8 +65,9 @@
     <PackageReference Include="Microsoft.VisualStudio.Threading" Version="$(MicrosoftVisualStudioThreadingVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.Utilities" Version="$(MicrosoftVisualStudioUtilitiesVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.Validation" Version="$(MicrosoftVisualStudioValidationVersion)" />
+    <PackageReference Include="Xunit.Combinatorial" Version="$(XunitCombinatorialVersion)" />
```
Comment (sharwell) [inline src/EditorFeatures/Test/Microsoft.CodeAnalysis.EditorFeatures.UnitTests.csproj]: 📝 This line is not necessary now that #41806 is merged.

**Pull request merged**